ipc/nodepart/arwenDriver.go

```diff
@@ -0,0 +1,308 @@
+}
+
+// RunSmartContractCreate sends a deploy request to Arwen and waits for the output
+func (driver *ArwenDriver) RunSmartContractCreate(input *vmcommon.ContractCreateInput) (*vmcommon.VMOutput, error) {
+	driver.nodeLogger.Trace("RunSmartContractCreate")
+
+	err := driver.restartArwenIfNecessary()
+	if err != nil {
+		return nil, common.WrapCriticalError(err)
+	}
+
+	request := common.NewMessageContractDeployRequest(input)
+	response, err := driver.part.StartLoop(request)
+	if err != nil {
+		driver.stopArwen()
+		return nil, common.WrapCriticalError(err)
+	}
+
+	typedResponse := response.(*common.MessageContractResponse)
+	return typedResponse.VMOutput, response.GetError()
+}
+
+// RunSmartContractCall sends an execution request to Arwen and waits for the output
+func (driver *ArwenDriver) RunSmartContractCall(input *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	driver.nodeLogger.Trace("RunSmartContractCall", "sc", input.RecipientAddr)
+
+	err := driver.restartArwenIfNecessary()
+	if err != nil {
+		return nil, common.WrapCriticalError(err)
+	}
+
+	request := common.NewMessageContractCallRequest(input)
+	response, err := driver.part.StartLoop(request)
+	if err != nil {
+		driver.nodeLogger.Error("RunSmartContractCall", "err", err)
+		driver.stopArwen()
+		return nil, common.WrapCriticalError(err)
+	}
+
+	typedResponse := response.(*common.MessageContractResponse)
+	return typedResponse.VMOutput, response.GetError()
+}
+
+// DiagnoseWait sends a diagnose message to Arwen
+func (driver *ArwenDriver) DiagnoseWait(milliseconds uint32) error {
+	err := driver.restartArwenIfNecessary()
+	if err != nil {
+		return common.WrapCriticalError(err)
+	}
+
+	request := common.NewMessageDiagnoseWaitRequest(milliseconds)
+	response, err := driver.part.StartLoop(request)
+	if err != nil {
+		driver.nodeLogger.Error("RunSmartContractCall", "err", err)
+		driver.stopArwen()
+		return common.WrapCriticalError(err)
+	}
+
+	return response.GetError()
+}
+
+// Close stops Arwen
+func (driver *ArwenDriver) Close() error {
+	err := driver.stopArwen()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (driver *ArwenDriver) stopArwen() error {
+	err := driver.command.Process.Kill()
+	driver.command.Process.Wait()
+	if err != nil {
+		driver.nodeLogger.Error("stopArwen error=%s", err)
+	}
+
+	return err
+}
+
+func (driver *ArwenDriver) continuouslyCopyArwenLogs(arwenStdout io.Reader, arwenStderr io.Reader) {
+	stdoutReader := bufio.NewReader(arwenStdout)
+	stderrReader := bufio.NewReader(arwenStderr)
+	arwenLog := driver.arwenLogRead
+
+	go func() {
+		for {
+			line, err := stdoutReader.ReadString('\n')
+			if err != nil {
+				break
+			}
+
+			line = strings.TrimSpace(line)
+			driver.nodeLogger.Info("ARWEN-OUT", "line", line)
+		}
+	}()
+
+	go func() {
+		for {
+			line, err := stderrReader.ReadString('\n')
+			if err != nil {
+				break
+			}
+
+			line = strings.TrimSpace(line)
+			driver.nodeLogger.Error("ARWEN-ERR", "line", line)
```

Comment:
why is this error ?

Reply:
Panics written to Arwen's `stderr` aren't received through the `pipe logger`, so we have to catch them somehow.